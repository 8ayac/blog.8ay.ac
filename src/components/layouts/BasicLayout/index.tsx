import { SiteFooter } from '@/src/components/widgets/SiteFooter';
import { SiteHeader } from '@/src/components/widgets/SiteHeader';
import { mq } from '@/src/shared/styles/mediaQuery';
import styled from '@emotion/styled';
import React, { ReactNode } from 'react';

const S = {
  ContentWrapperDiv: styled.div`
    max-width: 1000px;
    padding: 3rem 10%;
    margin: 3rem auto;
    font-size: 1.6rem;
    word-break: break-word;
    background-color: white;
    border-radius: 5px;

    img {
      max-width: 100%;
    }

    ${mq('sm')} {
      padding: 3rem 7.5%;
      margin: 0;
      font-size: 1.4rem;
      border-radius: 0;
    }
  `,
};

export const BasicLayout: React.FC<{ children?: ReactNode }> = ({
  children,
}) => (
  <>
    <SiteHeader />
    <S.ContentWrapperDiv>
      {children}
      <SiteFooter />
    </S.ContentWrapperDiv>
  </>
);
