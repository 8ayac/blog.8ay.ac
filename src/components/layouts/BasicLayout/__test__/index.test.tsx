import { BasicLayout } from '@/src/components/layouts/BasicLayout';
import { mount } from 'enzyme';
import React from 'react';

describe('BasicLayout', () => {
  const wrapper = mount(<BasicLayout />);

  test('is rendered correctly to match the snapshot', () => {
    wrapper.setProps({ children: <b>children</b> });
    expect(wrapper).toMatchSnapshot();
  });

  test('renders proper children', () => {
    wrapper.setProps({ children: <h1>THIS IS A TEST</h1> });
    expect(
      wrapper.find('ContentWrapperDiv').children().first().find('h1').text()
    ).toBe('THIS IS A TEST');
  });

  describe('has proper style rules', () => {
    test('in ContentWrapperDiv', () => {
      const wContentWrapperDiv = wrapper.find('ContentWrapperDiv');

      expect(wContentWrapperDiv).toHaveStyleRule('max-width', '1000px');
      expect(wContentWrapperDiv).toHaveStyleRule('padding', '3rem 10%');
      expect(wContentWrapperDiv).toHaveStyleRule('margin', '3rem auto');
      expect(wContentWrapperDiv).toHaveStyleRule('background-color', 'white');
      expect(wContentWrapperDiv).toHaveStyleRule('border-radius', '5px');

      expect(wContentWrapperDiv).toHaveStyleRule('max-width', '100%', {
        target: 'img',
      });
    });
  });
});
