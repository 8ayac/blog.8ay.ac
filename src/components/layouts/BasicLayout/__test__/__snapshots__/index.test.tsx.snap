// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicLayout is rendered correctly to match the snapshot 1`] = `
.emotion-0 {
  -webkit-position: sticky;
  position: sticky;
  top: 0;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 5rem;
  padding: 0.5rem 1.5rem;
  background-color: hsla(148, 72%, 30%);
  -webkit-filter: drop-shadow(0 0 2px hsla(148, 72%, 20%));
  filter: drop-shadow(0 0 2px hsla(148, 72%, 20%));
  opacity: 0.9;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  color: white;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 20%)
      );
  filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 20%)
      );
  opacity: 1;
}

.emotion-6 {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-6:hover {
  color: hsla(60, 72%, 60%);
}

.emotion-8 {
  font-size: 2.2rem;
}

.emotion-10 {
  margin-left: 0.5rem;
  font-size: 1.7rem;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 20%)
      );
  filter: drop-shadow(
        2px 2px 1px hsla(148, 72%, 20%)
      );
  opacity: 1;
}

.emotion-14 {
  margin: 0 0.5rem;
}

.emotion-16 {
  font-size: 2.2rem;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-16:hover {
  color: hsla(60, 72%, 60%);
}

.emotion-18 {
  margin-left: 0.5rem;
}

.emotion-20 {
  max-width: 1000px;
  padding: 3rem 10%;
  margin: 3rem auto;
  font-size: 1.6rem;
  word-break: break-word;
  background-color: white;
  border-radius: 5px;
}

.emotion-20 img {
  max-width: 100%;
}

.emotion-22 {
  padding: 3rem 0;
  margin-top: 10rem;
  font-size: 1.5rem;
  text-align: center;
}

.emotion-26 {
  color: hsla(0, 0%, 40%);
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "blue": Object {
          "base": "hsla(212, 72%, 60%)",
          "dark": "hsla(212, 72%, 50%)",
          "dark2": "hsla(212, 72%, 40%)",
          "dark3": "hsla(212, 72%, 30%)",
          "dark4": "hsla(212, 72%, 20%)",
          "light": "hsla(212, 72%, 70%)",
          "light2": "hsla(212, 72%, 80%)",
          "light3": "hsla(212, 72%, 90%)",
        },
        "border": Object {
          "primary": "hsla(0, 0%, 50%)",
          "primaryLight": "hsla(0, 0%, 75%)",
        },
        "green": Object {
          "base": "hsla(148, 72%, 60%)",
          "dark": "hsla(148, 72%, 50%)",
          "dark2": "hsla(148, 72%, 40%)",
          "dark3": "hsla(148, 72%, 30%)",
          "dark4": "hsla(148, 72%, 20%)",
          "light": "hsla(148, 72%, 70%)",
          "light2": "hsla(148, 72%, 80%)",
          "light3": "hsla(148, 72%, 90%)",
        },
        "primary": "hsla(212, 72%, 60%)",
        "primaryDark": "hsla(212, 72%, 50%)",
        "primaryLight": "hsla(212, 72%, 70%)",
        "text": Object {
          "accent": "hsla(148, 72%, 50%)",
          "primary": "hsla(0, 0%, 15%)",
          "primaryLight": "hsla(0, 0%, 40%)",
          "secondary": "hsla(212, 72%, 40%)",
        },
        "yellow": Object {
          "base": "hsla(60, 72%, 60%)",
          "dark": "hsla(60, 72%, 50%)",
          "dark2": "hsla(60, 72%, 40%)",
          "dark3": "hsla(60, 72%, 30%)",
          "dark4": "hsla(60, 72%, 20%)",
          "light": "hsla(60, 72%, 70%)",
          "light2": "hsla(60, 72%, 80%)",
          "light3": "hsla(60, 72%, 90%)",
        },
      },
    }
  }
>
  <BasicLayout>
    <SiteHeader>
      <Header>
        <header
          className="emotion-0 emotion-1"
        >
          <HeaderInner>
            <nav
              className="emotion-2 emotion-3"
            >
              <S-TitleWrapper>
                <div
                  className="emotion-4 emotion-5"
                >
                  <BlogTitle
                    subtitle="8ayac's blog"
                    title="もしくはこれ"
                  >
                    <Link
                      href="/"
                      passHref={true}
                    >
                      <LogoWrapperAnchor
                        href="/"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                      >
                        <a
                          className="emotion-6 emotion-7"
                          href="/"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                          <Logo
                            subtitle="8ayac's blog"
                            title="もしくはこれ"
                          >
                            <TitleSpan>
                              <span
                                className="emotion-8 emotion-9"
                              >
                                もしくはこれ
                              </span>
                            </TitleSpan>
                            <SubtitleSpan>
                              <span
                                className="emotion-10 emotion-11"
                              >
                                8ayac's blog
                              </span>
                            </SubtitleSpan>
                          </Logo>
                        </a>
                      </LogoWrapperAnchor>
                    </Link>
                  </BlogTitle>
                </div>
              </S-TitleWrapper>
              <S-NavWrapper>
                <div
                  className="emotion-12 emotion-13"
                >
                  <SocialLinkWithIcon
                    description="8ayac"
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f099",
                          "M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z",
                        ],
                        "iconName": "twitter",
                        "prefix": "fab",
                      }
                    }
                    url="http://twitter.com/8ayac"
                  >
                    <ComponentWrapperDiv>
                      <div
                        className="emotion-14 emotion-15"
                      >
                        <SocialLinkAnchor
                          href="http://twitter.com/8ayac"
                          rel="noopener"
                          target="_blank"
                        >
                          <a
                            className="emotion-16 emotion-17"
                            href="http://twitter.com/8ayac"
                            rel="noopener"
                            target="_blank"
                          >
                            <FontAwesomeIcon
                              border={false}
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    512,
                                    512,
                                    Array [],
                                    "f099",
                                    "M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z",
                                  ],
                                  "iconName": "twitter",
                                  "prefix": "fab",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-twitter fa-w-16 "
                                data-icon="twitter"
                                data-prefix="fab"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 512 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                            <LinkDescriptionSpan>
                              <span
                                className="emotion-18 emotion-19"
                              >
                                8ayac
                              </span>
                            </LinkDescriptionSpan>
                          </a>
                        </SocialLinkAnchor>
                      </div>
                    </ComponentWrapperDiv>
                  </SocialLinkWithIcon>
                </div>
              </S-NavWrapper>
            </nav>
          </HeaderInner>
        </header>
      </Header>
    </SiteHeader>
    <ContentWrapperDiv>
      <div
        className="emotion-20 emotion-21"
      >
        <b>
          children
        </b>
        <SiteFooter>
          <Footer>
            <footer
              className="emotion-22 emotion-23"
            >
              <CopyRightParagraph>
                <p
                  className="emotion-24 emotion-25"
                >
                  © 2021 blog.8ay.ac All Rights Reserved.
                </p>
              </CopyRightParagraph>
              <AboutGAParagraph>
                <p
                  className="emotion-26 emotion-27"
                >
                  This website uses Google Analytics, a web analytics service by Google. Google Analytics uses cookies, which are placed on your computer to help the website analyze how you use the site. You can prevent the the collection of your data by disabling cookies in your browser.
                </p>
              </AboutGAParagraph>
            </footer>
          </Footer>
        </SiteFooter>
      </div>
    </ContentWrapperDiv>
  </BasicLayout>
</ThemeProvider>
`;
